### PR TITLE
Add config file authentication to oci object storage state store

### PR DESCRIPTION
**Pull request by lucasjellema**
# Description

A very common way for providing configuration settings to tools (CLI, Terraform, Ansible, SDKs) and applications working with Oracle Cloud Infrastructure is through a configuration file (typically called config and located in ~/,oci/config) that contains the tenancy details and credentials for connecting to OCI (APIs). With this PR, the OCI ObjectStorage state store allows this type of configuration method. 

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[1428]_
https://github.com/dapr/components-contrib/issues/1428

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [X] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[2073]_

### CODE REVIEW

File: state/oci/objectstorage/objectstorage.go [hunk 1]
```diff
@@ -147,38 +175,94 @@ func NewOCIObjectStorageStore(logger logger.Logger) *StateStore {
...
+}
+
+func getConfigFilePath(meta map[string]string) (value string, err error) {
+	value, _ = getValue(meta, configFilePathKey, false)
+	if strings.HasPrefix(value, "~/") {
+		usr, _ := user.Current()
+		dir := usr.HomeDir
+		value = filepath.Join(dir, value[2:])
+	}
+	if value != "" {
```
Comment (daixiang0) [inline state/oci/objectstorage/objectstorage.go:207]: I prefer to use the absolute path here in case link issues.

Comment (lucasjellema) [reply state/oci/objectstorage/objectstorage.go:207]: Hi @daixiang0 , I am afraid I do not understand your comment. Do you mean to say that ~/ should not be supported at all? For clarity's sake, only accept absolute path values?

Comment (daixiang0) [reply state/oci/objectstorage/objectstorage.go:207]: I mean use an absolute clean path, not include relative path for clear.

Comment (lucasjellema) [reply state/oci/objectstorage/objectstorage.go:207]: I would like to understand you exactly right. Do you suggest to not allow ~/ as a shorthand for /home/<user>?

Comment (daixiang0) [reply state/oci/objectstorage/objectstorage.go:207]: You got it.

Comment (yaron2) [reply state/oci/objectstorage/objectstorage.go:207]: Ping @lucasjellema

Comment (lucasjellema) [reply state/oci/objectstorage/objectstorage.go:207]: I will fix this in the next commit; note: I need to update the documentation to reflect this change.

**Comment by lucasjellema** [comment]
Fixed documentation in https://github.com/dapr/docs/pull/2165

**Comment by yaron2** [comment]
Thanks @lucasjellema. Were you able to test the component to ensure no regressions? asking because we don't (yet) have conformance tests for it.

**Comment by lucasjellema** [comment]
> Thanks @lucasjellema. Were you able to test the component to ensure no regressions? asking because we don't (yet) have conformance tests for it.

I did and there is no regression. I use a local integration test , that I will commit as well (however - you will need an OCI account to run it). Do you have instructions on creating a conformance test? I am willing to try to create one.

**Comment by daixiang0** [comment]
You can refer to other conformance tests to create it.

**Review by daixiang0 — APPROVED** [review]
LGTM overall.

/cc @yaron2 as it is still alpha, we can merge it and add conformance tests later, WDYT?

**Comment by yaron2** [comment]
> LGTM overall.
> 
> /cc @yaron2 as it is still alpha, we can merge it and add conformance tests later, WDYT?

Agree

**Comment by yaron2** [comment]
Thanks for the contribution @lucasjellema

**Comment by daixiang0** [comment]
@lucasjellema please create an issue to track conformance tests.

**Comment by lucasjellema** [comment]
> @lucasjellema please create an issue to track conformance tests.

Done: https://github.com/dapr/components-contrib/issues/1475

**Pull request merged**